src/main/java/christmas/domain/Orders.java

```diff
@@ -0,0 +1,79 @@
+package christmas.domain;
+
+import christmas.dto.OrderMenuDto;
+
+import java.util.*;
+
+public class Orders {
+    private static final int MAXIMUM_ORDER_SIZE = 20;
+
+    private final Map<Menu, Integer> orders;
+
+    public Orders(List<OrderMenuDto> orders) {
+        validateOrderMenu(orders);
+        validateDuplicateMenu(orders);
+        validateOrderSize(orders);
+        validateOrderOnlyDrink(orders);
+        this.orders = initOrders(orders);
+    }
+
+    private void validateOrderMenu(List<OrderMenuDto> orders) {
+        for (OrderMenuDto order : orders) {
+            Menu.of(order.getMenuName());
+        }
+    }
+
+    private void validateDuplicateMenu(List<OrderMenuDto> orders) {
+        long menuCount = orders.stream()
+                .map(orderMenuDto -> orderMenuDto.getMenuName())
+                .distinct()
+                .count();
+        if (menuCount != orders.size()) {
+            throw new IllegalArgumentException(ErrorMessage.INVALID_MENU.get());
+        }
+    }
+
+    private void validateOrderSize(List<OrderMenuDto> orders) {
+        int orderSize = 0;
+        for (OrderMenuDto orderMenuDto : orders) {
+            orderSize += orderMenuDto.getTotal();
+            if (orderSize > MAXIMUM_ORDER_SIZE) {
+                throw new IllegalArgumentException(ErrorMessage.OVER_MAXIMUM_ORDER.get());
+            }
+        }
+    }
+
+    private void validateOrderOnlyDrink(List<OrderMenuDto> orders) {
+        int drinkCount = 0;
+        for (OrderMenuDto orderMenuDto : orders) {
+            Menu menu = Menu.of(orderMenuDto.getMenuName());
+            if (menu.getMenuType().equals(MenuType.음료)) {
+                drinkCount += 1;
+            }
+        }
+        if (drinkCount == orders.size()) {
+            throw new IllegalArgumentException(ErrorMessage.ONLY_DRINK.get());
+        }
+    }
```

Comment:
```suggestion
    private void validateOrderOnlyDrink(List<OrderMenuDto> orders) {
        long drinkCount = orders.stream()
                     .map(orderMenuDto -> Menu.of(orderMenuDto.getMenuName()))
                     .filter(menu -> menu.getMenuType().equals(MenuType.음료))
                     .count();

        if (drinkCount == orders.size()) {
            throw new IllegalArgumentException(ErrorMessage.ONLY_DRINK.get());
        }
}
```

stream을 사용하시면 더 간결하게 표현할 수 있을 것 같아요 :)